erpnext/hr/doctype/attendance_request/attendance_request.py

```diff
@@ -5,6 +5,62 @@
 from __future__ import unicode_literals
 import frappe
 from frappe.model.document import Document
+from frappe import _
+from frappe.utils import getdate, date_diff, add_days, nowdate
+from erpnext.hr.doctype.employee.employee import is_holiday
 
 class AttendanceRequest(Document):
-	pass
+	def validate(self):
+		self.validate_date()
+
+	def on_submit(self):
+		self.create_attendance()
+
+	def on_cancel(self):
+		attendance_list = frappe.get_list("Attendance", {'employee': self.employee, 'attendance_request': self.name})
+		if attendance_list:
+			for attendance in attendance_list:
+				attendance_obj = frappe.get_doc("Attendance", attendance['name'])
+				attendance_obj.cancel()
+
+	def validate_date(self):
+		date_of_joining = frappe.db.get_value("Employee", self.employee, "date_of_joining")
```

Comment:
We can do this in one query:
date_of_joining, relieving_date = frappe.db.get_value("Employee", self.employee, ["date_of_joining", "relieving_date"])